src/web-components/vf-score.js

```diff
@@ -0,0 +1,137 @@
+    this.stavesPerLine = 1; 
+
+    this.systemsAdded = 0; // Counter that keeps track of how many systems have dispatched 
+                           // events signalling that they are ready to be drawn. When the
+                           // number of children matches this counter, the entire score
+                           // is ready to be drawn.
+                           // FOR THIS PR: a vf-score can only have one vf-system. 
+
+    this.attachShadow({ mode:'open' });
+
+    // The 'systemCreated' event is dispatched by a vf-systen when it has finished creating 
+    // and adding its staves. vf-score listens to this event so that it can add that it can 
+    // detect when the vf-system is ready to be drawn.
+    this.addEventListener('systemCreated', this.systemCreated);
+
+    // These events are dispatched by the corresponding elements (ex: vfVoiceReady is dispatched
+    // by a vf-voice) when they are added to the DOM. vf-score listens to these events so that it 
+    // can set the elements' Factory and/or Registry instances, since these are shared by a 
+    // vf-score and all its children to maintain the same render queue. 
+    this.addEventListener('vfVoiceReady', this.setFactory);
+    this.addEventListener('vfStaveReady', this.setFactory);
+    this.addEventListener('vfStaveReady', this.setRegistry);
+    this.addEventListener('vfSystemReady', this.setFactory);
+  }
+
+  connectedCallback() {
+    this.x = parseInt(this.getAttribute('x')) || this.x;
+    this.y = parseInt(this.getAttribute('y')) || this.y;
+    this.width = parseInt(this.getAttribute('width')) || this.width;
+    this.height = parseInt(this.getAttribute('height')) || this.height;
+
+    this.rendererType = this.getAttribute('renderer'); 
+
+    this.setupVexflow();
+    this.setupFactory();
+
+    // vf-score listens to the slotchange event so that it can detect its system and set it up accordingly
+    this.shadowRoot.querySelector('slot').addEventListener('slotchange', this.registerSystem);
+  }
+
+  disconnectedCallback() {
+    this.shadowRoot.querySelector('slot').removeEventListener('slotchange', this.registerSystem);
+  }
+
+  /**
+   * Sets up the renderer, context, and registry for this component.
+   */
+  setupVexflow() {
+    // Default to svg renderer if not specified
+    const element = (this.rendererType === 'canvas') ? document.createElement('canvas') : document.createElement('div');
+    element.id = 'vf-score';
+    element.appendChild(document.createElement('slot'));
+    this.shadowRoot.appendChild(element);
+
+    if (this.rendererType === 'canvas') {
+      this.renderer = new Vex.Flow.Renderer(element, Vex.Flow.Renderer.Backends.CANVAS);
+    } else { 
+      this.renderer = new Vex.Flow.Renderer(element, Vex.Flow.Renderer.Backends.SVG);
+    }
+
+    this.renderer.resize(this.width, this.height);
+    this.context = this.renderer.getContext();
+    this.registry = new Vex.Flow.Registry();
```

Comment:
It'd be good to initialize all fields in the constructor, for documentation purposes at least. It makes it easier to scan and tell the shape of the class.